utilities/transactions/write_unprepared_txn_db.cc

```diff
@@ -270,12 +271,11 @@ Status WriteUnpreparedTxnDB::Initialize(
       ordered_seq_cnt[seq] = cnt;
       assert(wupt->unprep_seqs_.count(seq) == 0);
       wupt->unprep_seqs_[seq] = cnt;
-      KeySetBuilder keyset_handler(wupt,
-                                   txn_db_options_.rollback_merge_operands);
-      s = batch_info.batch_->Iterate(&keyset_handler);
+
+      s = wupt->RebuildFromWriteBatch(batch_info.batch_);
```

Comment:
It seems that we need a more optimized approach for recovery to avoid OOM during recovery of large transactions. The current recovery process rebuilds the entire write batch into recovered_trx->batches_ and then track the keys and delete the batch. Since wup enables the users to insert txns that don't fit into the memory, this could cause an issue during recovery of such large transactions.
Since handling it doesn't involve much risk, we can leave it as TODO and get back to it when more complicated part of the project are finished.

Reply:
Yeah, I think I documented that here:

https://github.com/facebook/rocksdb/blob/cd2520361d38ef3556d3bda479fd7a4caa0d1168/db/db_impl/db_impl.h#L639-L641